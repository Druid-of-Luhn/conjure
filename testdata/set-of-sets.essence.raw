Spec
  { language = "Essence"
  , version = [ 2
              , 0
              ]
  , topLevelBindings = [ ( Find
                         , "x"
                         , DomainSet
                             { size = Just ( ValueInteger 6 )
                             , minSize = Nothing
                             , maxSize = Nothing
                             , attrDontCare = False
                             , element = DomainSet
                                           { size = Just ( ValueInteger 4 )
                                           , minSize = Nothing
                                           , maxSize = Nothing
                                           , attrDontCare = False
                                           , element = DomainIntegerFromTo (
                                                         Just ( ValueInteger 0 ) ) (
                                                         Just ( ValueInteger 9 ) )
                                           , representation = Nothing
                                           }
                             , representation = Nothing
                             }
                         )
                       , ( Find
                         , "y"
                         , DomainSet
                             { size = Just ( ValueInteger 4 )
                             , minSize = Nothing
                             , maxSize = Nothing
                             , attrDontCare = False
                             , element = DomainIntegerFromTo (
                                           Just ( ValueInteger 2 ) ) (
                                           Just ( ValueInteger 7 ) )
                             , representation = Nothing
                             }
                         )
                       ]
  , topLevelWheres = []
  , objective = Nothing
  , constraints = [ GenericNode
                      Elem
                      [ Identifier "y"
                      , Identifier "x"
                      ]
                  ]
  , metadata = []
  }