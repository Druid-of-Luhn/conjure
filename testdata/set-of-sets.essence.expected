language Essence 2.0

find x : set (size 6) of set (size 4) of int(0..9)
find y : set (size 4) of int(2..7)

such that
    y elem x
